paasta_tools/docker_wrapper.py

```diff
@@ -330,4 +363,27 @@ def main(argv=None):
         except Exception as e:
             print(f'Unhandled exception in add_firewall: {e}', file=sys.stderr)
 
+    # Would probably also be based on some configuration so that
+    # the feature can be toggled for rollout / rollback
+    if can_add_ip_address(argv):
+        try:
+            # Need cross-process locking semantics to avoid race condition where
+            # multiple paasta invocations try to acquire the same IP address
+
+            # get_lock()
+
+            ip_addr = get_ip_for_container()
+            # Only add the --ip argument if we actually got an IP
+            if ip_addr:
+                argv = add_argument(argv, f'--ip={ip_addr}')
```

Comment:
Does this work with mesos? From docker docs, the --ip is supported only for user-defined networks.

There is a similar precedent in paasta code where --mac-addres is added  in paasta firewall: https://github.com/Yelp/paasta/blob/master/paasta_tools/docker_wrapper.py#L289

But I don't think the IP address can be pinned unless mesos runs a user-defined network.

Reply:
Hrmmm I didn’t consider any meals implications. Does it matter? We are completely controlling the `docker run` Invocation and the docker daemon, and media currently doesn’t have any influence on what IP address is chosen (it’s randomly assigned by docker run)

We would need to update our puppet docker daemon to use a custom bridge network, which I think should be fine, but I don’t really have any way of knowing how reasonable that is